### PR TITLE
Expose charsets as Hash in normalized params

### PR DESCRIPTION
**Pull request by heidi**
Based on https://github.com/ygnessin/griddler/pull/4, in tandem with https://github.com/apartmentlist/griddler/pull/1

Parse the JSON string and return a Hash with symbol keys
for ease of use by Griddler later on.

### CODE REVIEW

File: spec/griddler/sendgrid/adapter_spec.rb [hunk 1]
```diff
@@ -129,13 +137,38 @@
...
+      charsets[field].should eq 'UTF-8'
+    end
+  end
+
+  it 'does not explode if charsets is not JSON-able' do
+    params = default_params.merge(charsets: 'This is not JSON')
+
+    normalize_params(params)[:charsets].should be_nil
+  end
   def default_params
```
Comment (tcollier) [inline spec/griddler/sendgrid/adapter_spec.rb]: 👾

File: spec/griddler/sendgrid/adapter_spec.rb [hunk 2]
```diff
@@ -14,6 +14,14 @@
       to: 'Hello World <hi@example.com>',
       cc: 'emily@example.com',
       from: 'There <there@example.com>',
+      charsets: {
+        to: 'UTF-8',
+        cc: 'UTF-8',
+        subject: 'UTF-8',
+        from: 'UTF-8',
+        html: 'UTF-8',
+        text: 'iso-8859-1'
```
Comment (tcollier) [inline spec/griddler/sendgrid/adapter_spec.rb]: You don't gain much in this test by having an exhaustive list of `charsets`. I suggest only having 2, with different values. Then you can do away with the loop in the spec.

File: spec/griddler/sendgrid/adapter_spec.rb [hunk 3]
```diff
@@ -129,13 +137,38 @@
...
+    normalize_params(params)[:charsets].should be_nil
+  end
   def default_params
     {
       text: 'hi',
       to: '"Mr Fugushima at Fugu, Inc" <hi@example.com>, Foo bar <foo@example.com>, Eichhörnchen <squirrel@example.com>, <no-name@example.com>',
       cc: 'cc@example.com',
       from: 'there@example.com',
       envelope: "{\"to\":[\"johny@example.com\"], \"from\": [\"there@example.com\"]}",
+      charsets: {
```
Comment (tcollier) [inline spec/griddler/sendgrid/adapter_spec.rb]: No specs are using this, so I suggest leaving it out.

File: lib/griddler/sendgrid/adapter.rb [hunk 4]
```diff
@@ -40,6 +41,15 @@ def bcc_from_envelope
         JSON.parse(params[:envelope])["to"] if params[:envelope].present?
       end
 
+      def charsets
+        if params[:charsets].present?
+          JSON.parse(params[:charsets]).symbolize_keys
+        end
+      rescue JSON::ParserError
+        nil
```
Comment (tcollier) [inline lib/griddler/sendgrid/adapter.rb]: Did you consider defaulting to a `{}`?

Comment (heidi) [reply lib/griddler/sendgrid/adapter.rb]: Not originally, but I agree: `{}` is better than `nil`.

File: spec/griddler/sendgrid/adapter_spec.rb (hunk 3 above)
Comment (heidi) [reply spec/griddler/sendgrid/adapter_spec.rb]: The `it 'returns the charsets as a hash'` spec is using it.

File: spec/griddler/sendgrid/adapter_spec.rb (hunk 2 above)
Comment (heidi) [reply spec/griddler/sendgrid/adapter_spec.rb]: 👍

**Pull request merged**